examples/ArduinoWiFi101/ArduinoWiFi101.ino

```diff
@@ -62,10 +66,6 @@ void loop() {
   }
 }
 
-void messageReceived(String topic, String payload, char * bytes, unsigned int length) {
-  Serial.print("incoming: ");
-  Serial.print(topic);
-  Serial.print(" - ");
-  Serial.print(payload);
-  Serial.println();
+void messageReceived(String topic, String payload) {
```

Comment:
All of the projects I did with students have been about sending an receiving numeric values. Therefore a simple `payload.toInt()` is sufficient and we never ran into any issues. Event if you use JSON you should not have a null termination in the payload. Thus, if somebody manages to construct a binary payload and wants to decode it, that person would have suffcient experience to use the `onAdvanvedMessage` callback.

The same story goes for the topics. We always used the `topic.equals("/fridge/on")` method to match the topic. You rarely need to split the topic, except it you transmit dynamic data in the topic, which I would anway not recommend.

What about a "Best Practices" section in the Readme?